gcc-cci/Dockerfile

```diff
@@ -0,0 +1,116 @@
+FROM ubuntu:xenial as base
+
+LABEL maintainer="Conan.io <info@conan.io>"
+
+ARG COMPILER_VERSION
+
+ENV PYENV_ROOT=/opt/pyenv \
+    PATH=/opt/pyenv/shims:/usr/local/bin:${PATH} \
+    CXX=/usr/bin/g++ \
+    CC=/usr/bin/gcc \
+    PYTHON_VERSION=3.8.2 \
+    CMAKE_VERSION=3.17.2 \
+    GCC_VERSION=$COMPILER_VERSION
+
+RUN apt-get -qq update \
+    && apt-get -qq install -y --no-install-recommends \
+       sudo \
+       build-essential \
+       wget \
+       git \
+       libc6-dev \
+       gcc \
+       libgmp-dev \
+       libmpfr-dev \
+       libmpc-dev \
+       nasm \
+       dh-autoreconf \
+       libffi-dev \
+       libssl-dev \
+       pkg-config \
+       subversion \
+       zlib1g-dev \
+       libbz2-dev \
+       libsqlite3-dev \
+       libreadline-dev \
+       xz-utils \
+       curl \
+       libncurses5-dev \
+       libncursesw5-dev \
+       liblzma-dev \
+       ca-certificates \
+       autoconf-archive \
+       flex \
```

Comment:
are you planning to pin the versions of those packages?

Reply:
No, pinning versions on Ubuntu is useless, as it doesn't keep old versions available for installation. We tried it in the past, but it caused us more problems than security and stability.

Reply:
Like the attitude 👍 So no issues with reproducibility?

Reply:
Well, you could create a builder/base image (and publish it for example to, say GitHub packages) and then only update compiler / conan.